evidence/pandres95/00005-2025-05-promote-rank-II.md

```diff
@@ -0,0 +1,84 @@
+
+To close this gap, I started by assessing the implementation state of `fungibles`, the set of traits that allowed achieving these capabilities. While regular inspection, mutation (issuing, burning), unbalances handling, and transferring were already implemented, two main traits were lacking formal implementation on `pallet-assets`: support for freezing and holding assets.
+
+Freezing assets is a capability that enables use cases like voting using non-native assets, staking, and vesting. This led me to implement the **[`pallet-assets-freezer`](https://github.com/paritytech/polkadot-sdk/pull/3951)**, a pluggable component for `pallet-assets` that extends the its base functionality by implementing the freezing of balances.
+
+In consequence, this implementation enabled the necessary support for implementing two additional components: [`pallets-asset-rewards`](https://github.com/paritytech/polkadot-sdk/pull/3926) (implemented by [@liamaharon](https://github.com/liamaharon), and already released), which allows (among others) incentivising the freezing (staking) of LP tokens, thus incentivising the usage of Asset Conversion, a key component in the Hub; but also **[`pallet-assets-vesting`](https://github.com/paritytech/polkadot-sdk/pull/7404)** (currently waiting for a secondary review), which provides a linear vesting mechanism for balances in assets accounts, analogous to the one existing in `pallet-vesting`. This is especially good for teams in the ecosystem (an example of this is vested acquisition of tokens in DAOs), grants and crowd-loans.
+
+Holding assets is the ability that an external component has to withhold a balance, in such a way it can be burned (slashed) by the system. By designing, and implementing the **[`pallet-assets-holder`](https://github.com/paritytech/polkadot-sdk/pull/4530)** component, I aimed to close the final gap in the `fungibles` implementation, a process that also involved [a major refactor](https://github.com/paritytech/polkadot-sdk/pull/4530#discussion_r1708418726) of the balances model for Assets, since the previous model was not fully compliant with the [specification](https://paritytech.github.io/polkadot-sdk/master/frame_support/traits/tokens/fungible/index.html), a clear proof of an _analytical improvement of a major component of the protocol_.
```

Comment:
Are these two used anywhere in prod?

Reply:
Or put differently, is there a plan by anyone to use `pallet-assets-vesting` and `pallet-assets-holder` once they are merged?